src/error-table.rkt

```diff
@@ -0,0 +1,49 @@
+#lang racket
+
+(require "points.rkt" "syntax/types.rkt" "core/localize.rkt" "timeline.rkt")
+
+(provide group-errors)
+
+(define (zip l1 l2) (map list l1 l2))
+
+(define (group-errors expr pcontext)
+  (define points (for/list ([(pt _) (in-pcontext pcontext)]) pt))
+  (define local-errors (car (compute-local-errors (list expr) (*context*))))
+
+  (define max-per-point (make-hash))
+  (define errpts-per-subexpr (make-hash))
+
+  (for ([pt points])
+    (hash-set! max-per-point pt (list 0 0)))
+
+  (for ([(sub-expr errors) local-errors])
+    (hash-set! errpts-per-subexpr sub-expr null)
+    (for ([pair (zip points errors)])
+      (define pt (car pair))
+      (define err (cadr pair))
+      (define old-error (hash-ref max-per-point pt))
+      (if (> err (cadr old-error))
+        (hash-set! max-per-point pt (list sub-expr err))
+        null)))
+  
+  (hash-set! errpts-per-subexpr #f null)
+
+  (for ([(k v) max-per-point])
+    (define sub-expr (car v))
+    (define err (cadr v))
+    (if (> err 2)
+      (hash-update! errpts-per-subexpr sub-expr (curry cons k))
+      (hash-update! errpts-per-subexpr #f (curry cons k))))
+  
+  ; (eprintf "[grouped errors]\n")
+  ; (for ([(k v) errpts-per-subexpr])
+  ;   (eprintf "~a, ~a \n" k v))
+  
+  ; (eprintf "\n[percentage-errored]\n")
+  (for ([(k v) errpts-per-subexpr])
```

Comment:
It's good practice to do iteration over `(in-hash errpts-per-subexpr)`, which both serves as a type hint and also is faster.